### ISSUE TITLE
The crate does not compile in OmniOS

### ISSUE THREAD

**Issue by mzanrosso** [Lokathor/tinyvec#125]
### Hi everyone, I have a VM with omnios and when I try to do cargo build on a project that has as an indirect dependency tinyvec goes into segmentation fault.

The main specifications of the VM are these: 
- OS: OmniOS v11 r151030cc i386
- CPU: Intel i5-5250U (1) @ 1.600GHz
- RAM: 1527MiB
- RUSTC: v1.47.0
- CARGO: v1.47.0
- GCC: 5.5.0

```
error: could not compile `tinyvec`.
Caused by:
  process didn't exit successfully: `rustc --crate-name tinyvec --edition=2018 /home/vagrant/.cargo/registry/src/github.com-1ecc6299db9ec823/tinyvec-1.0.1/src/lib.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts --crate-type lib --emit=dep-info,metadata,link -C embed-bitcode=no -C debuginfo=2 --cfg 'feature="alloc"' --cfg 'feature="default"' --cfg 'feature="tinyvec_macros"' -C metadata=4033b7c8c11b21a1 -C extra-filename=-4033b7c8c11b21a1 --out-dir [PATH OVERWRITTEN] -L dependency=[PATH OVERWRITTEN] --extern tinyvec_macros=[PATH OVERWRITTEN] --cap-lints allow` (signal: 11, SIGSEGV: invalid memory reference)
```
- [PATH OVERWRITTEN] = real path replaced for privacy

**Comment by Lokathor** [comment]
This looks like a bug in `rustc` itself, not a bug in `tinyvec`.

I hate to say that it's "not my problem", but you have to file a bug with the [rust-lang/rust](https://github.com/rust-lang/rust/issues) issue tracker to get help with this one. There's nothing that the `tinyvec` crate can do about a segmentation fault.

**Comment by mzanrosso** [comment]
> This looks like a bug in `rustc` itself, not a bug in `tinyvec`.
> 
> I hate to say that it's "not my problem", but you have to file a bug with the [rust-lang/rust](https://github.com/rust-lang/rust/issues) issue tracker to get help with this one. There's nothing that the `tinyvec` crate can do about a segmentation fault.

Thanks anyway for the answer, I opened an issue as you suggested;  you can follow the progress here https://github.com/rust-lang/rust/issues/79558

**Comment by Lokathor** [comment]
It appears that this was fixed in 1.49

@mzanrosso can you confirm?

**Comment by mzanrosso** [comment]
Yes, now it's fixed
